consumer.go

```diff
@@ -213,20 +213,22 @@ func (c *Consumer) run() {
 				// Therefore we check the length of the channel and await for it to reach 0. If for some reason
 				// it fails to drain after a number of attempts we continue on and allow the messages to simply timeout
 				// and be reqeueued by the nsqd server.
-				attempts := 0
-				for len(cm.CmdChan) > 0 {
-					// If we've tried to allow the messages to flush and they've failed after
-					// 3 attempts we give up and let the nsqd instance timeout and requeue for us.
-					if attempts > 2 {
-						log.Printf("failed to requeue %d messages during orderly shutdown, allow them to timeout and requeue", len(cm.CmdChan))
-						break
+				go func(cm connMeta) {
```

Comment:
Do we need a wait group or some other coordination here?  Looks to me like the calling func is free to exit and `c.shutJoin.Done()` can get called before they have a chance to finish.

Reply:
good catch